risc0/zkp/src/verify/mod.rs

```diff
@@ -33,6 +33,7 @@ use crate::{
 };
 
 #[derive(PartialEq)]
+#[non_exhaustive]
```

Comment:
What does this do?

Reply:
It indicates that more enum variants (i.e. more error conditions) may be added in the future, and so Rust will give a compile error to developers that try to write an exhaustive `match` statement in their code. Adding this attribute prevents adding a new enum variant from being a breaking change, since all callers are required to handle the "unknown variant" case.

Reply:
Wouldn't we want new enum variants to represent a breaking change?

Reply:
My intuition in this case is "no". If your intuition is "yes", then I'll go ahead and remove this one.

Reply:
I'm not sure, let's leave it and see what happens over time